### PR TITLE
remove old config from INC 2.x API

### PR DESCRIPTION
**Pull request by violetch24**
## Type of Change

API changed or not

## Description

JIRA: https://jira.devtools.intel.com/browse/ILITV-2751

## Expected Behavior & Potential Risk

N/A

## How has this PR been tested?

unit test

## Dependency Change?

N/A

### CODE REVIEW

**Comment by PenghuiCheng** [comment]
extension test：https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/4813/

**Comment by violetch24** [comment]
extension test rebuild: https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/4838/

File: .azure-pipelines/scripts/codeScan/pyspelling/inc_dict.txt [hunk 1]
```diff
@@ -1331,6 +1331,8 @@ OTHpart
 Ott
 oup
 outdir
+Outlier
```
Comment (ftian1) [inline .azure-pipelines/scripts/codeScan/pyspelling/inc_dict.txt:1334]: why this word is putting into the skip word list? Apparently the word is wrong, you should use "The outlier" or "An outlier".

File: .azure-pipelines/scripts/codeScan/pyspelling/inc_dict.txt [hunk 2]
```diff
@@ -2597,3 +2599,4 @@ pc
 cdrdv
 NeuralCompressor
 zp
+TensorflowTopK
```
Comment (ftian1) [inline .azure-pipelines/scripts/codeScan/pyspelling/inc_dict.txt:2602]: this word is also quite weird.

File: examples/helloworld/tf_example1/README.md [hunk 3]
```diff
@@ -30,12 +30,14 @@ python test.py --dataset_location=/path/to/imagenet/
 ### 2. Introduction 
 We only need to add the following lines for quantization to create an int8 model.
 ```python
+    top1 = TensorflowTopK(k=1)
```
Comment (ftian1) [inline examples/helloworld/tf_example1/README.md:33]: where the TensorflowTopk comes from? such function shouldn't expose to user

File: examples/helloworld/tf_example1/README.md [hunk 4]
```diff
@@ -30,12 +30,14 @@ python test.py --dataset_location=/path/to/imagenet/
...
     config = PostTrainingQuantConfig(calibration_sampling_size=[20])
     q_model = fit(
         model="./mobilenet_v1_1.0_224_frozen.pb",
         conf=config,
         calib_dataloader=calib_dataloader,
-        eval_dataloader=eval_dataloader)
+        eval_dataloader=eval_dataloader,
+        eval_metric=top1)
```
Comment (ftian1) [inline examples/helloworld/tf_example1/README.md:40]: why this issue just gets raised now?

File: examples/helloworld/tf_example1/test.py [hunk 5]
```diff
@@ -6,6 +6,7 @@
 from neural_compressor.data import DefaultDataLoader
 from neural_compressor.quantization import fit
 from neural_compressor.config import PostTrainingQuantConfig
+from neural_compressor.metric import TensorflowTopK
```
Comment (ftian1) [inline examples/helloworld/tf_example1/test.py:9]: why we doesn't support below usage @PenghuiCheng 

from neural_compressor import Metric
eval_metric = Metric('topk', ...)

**Review by PenghuiCheng — APPROVED** [review]

File: examples/helloworld/tf_example1/test.py (hunk 5 above)
Comment (PenghuiCheng) [reply examples/helloworld/tf_example1/test.py:9]: We support method above

File: examples/helloworld/tf_example1/README.md (hunk 4 above)
Comment (PenghuiCheng) [reply examples/helloworld/tf_example1/README.md:40]: in pythonic_config, there is default value for eval_metric.

File: neural_compressor/benchmark.py [hunk 6]
```diff
@@ -158,24 +153,21 @@ def __init__(self, conf):
         self._results = {}
         assert isinstance(conf, BenchmarkConfig), \
             "The config object should be config.BenchmarkConfig, not {}".format(type(conf))
-        conf = Config(quantization=None, benchmark=conf, pruning=None, distillation=None, nas=None)
-        self.conf = BenchmarkConf()
-        self.conf.map_pyconfig_to_cfg(conf)
-        if self.conf.usr_cfg.model.framework != 'NA':
-            self.framework = self.conf.usr_cfg.model.framework.lower()
+        self.conf = Config(quantization=None, benchmark=conf, pruning=None, distillation=None, nas=None)
```
Comment (ftian1) [inline neural_compressor/benchmark.py:156]: what's the Config class for? we have had many Config classes, why still need an extra "Config" class? if it's internal class for orchestration, we should use _Config but not Config

File: neural_compressor/compression/callbacks.py [hunk 7]
```diff
@@ -200,7 +201,7 @@ def model(self, user_model):
             user_model: user are supported to set model from original framework model format
                         (eg, tensorflow frozen_pb or path to a saved model),
                         but not recommended. Best practice is to set from a initialized
-                        neural_compressor.experimental.common.Model.
+                        neural_compressor.model.Model.
```
Comment (ftian1) [inline neural_compressor/compression/callbacks.py:204]: neual_compressor.model.Model is quite weird, it's better to use neural_compressor.Model

File: neural_compressor/compression/callbacks.py [hunk 8]
```diff
@@ -213,22 +214,28 @@ def model(self, user_model):
...
-                if self.cfg.model.backend == "default":
+            except Exception as e:
+                pass
+
+        if self.framework == "pytorch":
+            try:
+                if self.cfg.quantization.backend == "default":
                     self.framework = "pytorch_fx"
-                elif self.cfg.model.backend == "ipex":
+                elif self.cfg.quantization.backend == "ipex":
```
Comment (ftian1) [inline neural_compressor/compression/callbacks.py:234]: coding style inconsistent. if you take a look above, it's self.conf but not self.cfg

File: neural_compressor/compression/callbacks.py [hunk 9]
```diff
@@ -257,45 +273,48 @@ def pre_process(self):
...
                 logger.warning("Only tensorflow, pytorch_fx is supported by MSE_v2 currently.")
         assert strategy in STRATEGIES, "Tuning strategy {} is NOT supported".format(strategy)
 
         _resume = None
         # check if interrupted tuning procedure exists. if yes, it will resume the
         # whole auto tune process.
-        self.resume_file = os.path.abspath(os.path.expanduser(self.cfg.tuning.workspace.resume)) \
-                           if self.cfg.tuning.workspace and self.cfg.tuning.workspace.resume else None
+        self.resume_file = os.path.abspath(os.path.expanduser(self.cfg.options.resume_from)) \
+                           if self.cfg.options.workspace and self.cfg.options.resume_from else None
```
Comment (ftian1) [inline neural_compressor/compression/callbacks.py:294]: self.cfg.options are deprecated one. we are using func to set those global variables. pls confirm if this right implementation

File: neural_compressor/contrib/strategy/sigopt.py [hunk 10]
```diff
@@ -73,19 +72,43 @@ def eval_func(model):
...
+        Args:
+            model: The FP32 model specified for low precision tuning.
+            conf: The Conf class instance includes all user configurations.
+            q_dataloader: Data loader for calibration, mandatory for post-training quantization.  Defaults to None.
+            q_func: Training function for quantization aware training. Defaults to None. Defaults to None.
+            eval_func: The evaluation function provided by user. This function takes model as parameter, and 
+                evaluation dataset and metrics should be encapsulated in this function implementation and 
+                outputs a higher-is-better accuracy scalar value.
+            eval_dataloader: Data loader for evaluation. Defaults to None.
+            eval_metric: Metric for evaluation. Defaults to None.
```
Comment (ftian1) [inline neural_compressor/contrib/strategy/sigopt.py:96]: pls double check if sigopt strategy could work with Experiemental API (1.x API)

Comment (ftian1) [reply neural_compressor/contrib/strategy/sigopt.py:96]: ok, I saw this strategy gets implemented twice ( in experimental dir and here)

**Review by ftian1 — DISMISSED** [review]
pls check the comments I made. some variables should be invisible for user. pls add _ before them

File: neural_compressor/compression/callbacks.py (hunk 9 above)
Comment (PenghuiCheng) [reply neural_compressor/compression/callbacks.py]: done

**Review by PenghuiCheng — APPROVED** [review]
